delivery_carrier_report_to_printer/models/delivery_carrier.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2025 Tecnativa - Carlos Lopez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import Command, fields, models
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = "delivery.carrier"
+
+    label_report_id = fields.Many2one(
+        comodel_name="ir.actions.report",
+        domain=[("model", "=", "delivery.carrier")],
+    )
+
+    def send_shipping(self, pickings):
+        domain = [("res_id", "in", pickings.ids), ("res_model", "=", pickings._name)]
+        # Capture the created attachments during the shipping process
+        current_attachments = self.env["ir.attachment"].search(domain, order="id")
+        res = super().send_shipping(pickings)
+        new_attachments = (
+            self.env["ir.attachment"].search(domain, order="id") - current_attachments
+        )
+        if self.label_report_id and new_attachments:
+            self._print_carrier_label(new_attachments)
+        return res
```

Comment:
I think we should avoid both searches for  cases where `label_report_id` is not set checking the field first
```suggestion
        if not self.label_report_id:
            return super().send_shipping(pickings)
        domain = [("res_id", "in", pickings.ids), ("res_model", "=", pickings._name)]
        # Capture the created attachments during the shipping process
        current_attachments = self.env["ir.attachment"].search(domain, order="id")
        res = super().send_shipping(pickings)
        new_attachments = (
            self.env["ir.attachment"].search(domain, order="id") - current_attachments
        )
        if new_attachments:
            self._print_carrier_label(new_attachments)
        return res
```

Reply:
Done, thanks.